ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -118,6 +120,39 @@ void* breaker(void *d) {
   return NULL;
 }
 
+EvalParams* callback_eval_params = NULL;
+
+void mem_alloc_callback(ObjectSpace space, AllocationAction action, int size) {
+    if (NULL == callback_eval_params)
+    {
+        return; // this is probably an error that has to do with multi-threaded usage. Error somehow -seanmakesgames
+    }
+
+    if (action == kAllocationActionAllocate)
+    {
+        callback_eval_params->current_mem += size;
+        
+        printf("\nALLOC %d: %zu", size, callback_eval_params->current_mem);
+    }
+    else if (action == kAllocationActionFree)
+    {
+        callback_eval_params->current_mem -= size;
```

Comment:
Yeah. Or.... realistically size_t is probably overkill: unless people want to set memory limits of over 2 gigs (!!), using an int (or int32_t to be safe) instead of size_t would probably work. int64_t (either long or long long depending on platform) if you want to pad it out more. Then you can dip into the negatives sanely.